mesheryctl/internal/cli/root/system/start.go

```diff
@@ -173,6 +179,22 @@ func start() error {
 			}
 		}
 
+		var endpoint meshkitutils.HostPort
+		endpoint.Address = "http://localhost"
+		tempPort, err := strconv.Atoi(userPort[len(userPort)-1])
+		if err != nil {
+			return err
+		}
+		endpoint.Port = int32(tempPort)
+
+		var mockEndpoint *meshkitutils.MockOptions
+		mockEndpoint = nil
+
+		res := meshkitutils.TcpCheck(&endpoint, mockEndpoint)
+		if res {
+			return errors.New("the endpoint is not accessible")
+		}
+
```

Comment:
LGTM, we're checking if any other services aren't already running on the defined port, right?

Reply:
Ideally, we are also invoking an open (unauthenticated) API on Meshery to verify that the endpoint is in fact Meshery and is healthy; e.g. `/api/server/version`. @mgfeller @kumarabd @Aisuko recommendations on:

1. a "/healthz" endpoint in Meshery Server?
2. a reusable approach in MeshKit for verifying an application response beyond an open TCP connection/endpoint?

Reply:
Ohh I have one tiny idea, can we add it as `runPortAvailabilityHealthCheck()` in `check.go`??

Thoughts?? @navendu-pottekkat @leecalcote

Reply:
Do we have such a health endpoint defined for Meshery components?